generator/src/test/java/com/scottlogic/deg/generator/cucumber/MatchingRegex.feature

```diff
@@ -466,44 +466,66 @@ Scenario: Running a 'matchingRegex' request alongside a granularTo constraint sh
-     Given foo is matching regex /[0-z]{2}/
+     Given foo is matching regex /[a-z]{2}/
        And foo is after 2018-10-10T00:00:00.000
-     Then the following data should be included in what is generated:
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
         | foo  |
-        | null |
+        | "aa" |
         | "ab" |
+        | "ac" |
+        | "ad" |
+        | "ae" |
 
 Scenario: Running a 'matchingRegex' request alongside a afterOrAt constraint should be successful
-      Given foo is matching regex /[a-z]{2}/
-        And foo is after or at 2018-10-10T00:00:00.000
-      Then the following data should be included in what is generated:
-        | foo  |
-        | null |
-        | "ab" |
+     Given foo is matching regex /[a-z]{2}/
+       And foo is after or at 2018-10-10T00:00:00.000
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo  |
+       | "aa" |
+       | "ab" |
+       | "ac" |
+       | "ad" |
+       | "ae" |
 
 Scenario: Running a 'matchingRegex' request alongside a before constraint should be successful
-      Given foo is matching regex /[a-z]{2}/
-        And foo is before 2018-10-10T00:00:00.000
-      Then the following data should be included in what is generated:
-        | foo  |
-        | null |
-        | "ab" |
+     Given foo is matching regex /[a-z]{2}/
+       And foo is before 2018-10-10T00:00:00.000
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo  |
+       | "aa" |
+       | "ab" |
+       | "ac" |
+       | "ad" |
+       | "ae" |
 
 Scenario: Running a 'matchingRegex' request alongside a beforeOrAt constraint should be successful
-      Given foo is matching regex /[a-z]{2}/
-        And foo is before or at 2018-10-10T00:00:00.000
-      Then the following data should be included in what is generated:
-        | foo  |
-        | null |
-        | "ab" |
+     Given foo is matching regex /[a-z]{2}/
+       And foo is before or at 2018-10-10T00:00:00.000
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo  |
+       | "aa" |
+       | "ab" |
+       | "ac" |
+       | "ad" |
+       | "ae" |
 
+# Defect 621 "Matching Regex with a complementary not matching regex results in the test failing when it finds the expected dat" related to this scenario
+@ignore
 Scenario: Running a 'matchingRegex' request with a not constraint should be successful
-     Given foo is matching regex /[0-9]{1}/
-       And foo is anything but matching regex /[0-1]{1}/
+     Given foo is matching regex /[a-j]{1}/
+       And foo is anything but matching regex /[a-b]{1}/
+       And the generator can generate at most 5 rows
      Then the following data should not be included in what is generated:
```

Comment:
why should c, d, e,f and g not be included in what is generated? I don't understand this, surely they should be valid values?

Reply:
I'd not realised that this was a not generated - I've corrected it now.